src/operator/src/statement.rs

```diff
@@ -343,6 +352,31 @@ impl StatementExecutor {
             "DATESTYLE" => set_datestyle(set_var.value, query_ctx)?,
 
             "CLIENT_ENCODING" => validate_client_encoding(set_var)?,
+            // TODO: write sqlness test for query timeout variables
+            // once the proper channel is configured in the test infra.
+            // The current sqlness test channel is default to Unknown.
+            "MAX_EXECUTION_TIME" => match query_ctx.channel() {
+                Channel::Mysql => set_query_timeout(set_var.value, query_ctx)?,
+                Channel::Postgres => {
+                    query_ctx.set_warning(format!("Unsupported set variable {}", var_name))
+                }
+                _ => {
+                    return NotSupportedSnafu {
+                        feat: format!("Unsupported set variable {}", var_name),
+                    }
+                    .fail()
+                }
+            },
+            "STATEMENT_TIMEOUT" => {
```

Comment:
Maybe we can always support these two variables without considering the channel. @sunng87 @MichaelScofield

Reply:
This will make testing easier, will make the change if this gets another vote.

Reply:
I'm thinking maybe it's better to move the protocol specific variables to their server implementations. For example, the `STATEMENT_TIMEOUT` is only valid in pg, then we can deal with it in the pg related mods in crate server. In doing so we can get rid of these `if`s to the `channel`. WDYT? @sunng87 @evenyag

Reply:
> the STATEMENT_TIMEOUT is only valid in pg, then we can deal with it in the pg related mods in crate server. In doing so we can get rid of these ifs to the channel.

Do we need to refactor the way we handle the variables?

Reply:
@evenyag yes, actually I think it's an opportunity to get rid of the ugly "federated.rs" in mysql mod.

Reply:
@MichaelScofield Could you create an issue for this? We may not address it in this PR.

Reply:
@evenyag https://github.com/GreptimeTeam/greptimedb/issues/1712